compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/jvm/FirJvmExternalDeclarationChecker.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.checkers.declaration.jvm
+
+import org.jetbrains.kotlin.fir.FirFakeSourceElementKind
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.checkers.declaration.FirMemberDeclarationChecker
+import org.jetbrains.kotlin.fir.analysis.checkers.declaration.FirModifierList
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.declarations.*
+import org.jetbrains.kotlin.lexer.KtTokens
+
+// TODO: Move this to different, JVM-specific module?
+object FirJvmExternalDeclarationChecker : FirMemberDeclarationChecker() {
+    override fun check(declaration: FirMemberDeclaration, context: CheckerContext, reporter: DiagnosticReporter) {
+        if (!declaration.isExternal) return
+        val source = declaration.source ?: return
+        if (source.kind is FirFakeSourceElementKind) return
+
+        // WRONG_MODIFIER_TARGET on external constructor is intentionally NOT covered in this checker.
+        if (declaration !is FirFunction<*>) {
+            val target = when (declaration) {
+                is FirProperty -> "property"
+                is FirRegularClass -> "class"
+                else -> "non-function declaration"
+            }
+            val modifierList = with(FirModifierList) { source.getModifierList() }
+            val externalModifier = modifierList?.modifiers?.firstOrNull { it.token == KtTokens.EXTERNAL_KEYWORD }
```

Comment:
Widely used pattern in checkes, mind extracting it into a util function like`FirDeclaration.getModifierOfType(tokenType: KtModifierKeywordToken)` ?

Reply:
Done, added a few more utility functions.